charmcraft/templates/init-kubernetes/tests/unit/test_charm.py.j2

```diff
@@ -3,21 +3,22 @@
 #
 # Learn more about testing at: https://juju.is/docs/sdk/testing
 
-import unittest
-
 import ops
 import ops.testing
+import pytest
+
 from charm import {{ class_name }}
 
 
-class TestCharm(unittest.TestCase):
-    def setUp(self):
-        self.harness = ops.testing.Harness({{ class_name }})
-        self.addCleanup(self.harness.cleanup)
-        self.harness.begin()
+@pytest.fixture
+def harness():
+    harness = ops.testing.Harness({{ class_name }})
+    harness.begin()
+    yield harness
+    harness.cleanup()
```

Comment:
Have you generated each of these templates and run tox against the created charm? If not, that would be good to do. I'm wondering partly because I think the formatter would want two blank lines between top-level `def`s.

Reply:
My apologies, I should have known better. All three profiles pass now

Reply:
Hmm... spread tests should have picked up any tox failures in the `full-lifecycle` test. Thanks for uncovering a testing flaw! https://github.com/canonical/charmcraft/issues/1553